aptos-move/framework/aptos-framework/sources/dkg_rounding.move

```diff
@@ -0,0 +1,289 @@
+module aptos_framework::dkg_rounding {
+    use std::option;
+    use std::option::Option;
+    use std::vector;
+    use aptos_std::fixed_point64;
+    use aptos_std::fixed_point64::FixedPoint64;
+    #[test_only]
+    use std::string::utf8;
+    #[test_only]
+    use aptos_std::debug;
+    #[test_only]
+    use aptos_framework::randomness;
+
+    friend aptos_framework::reconfiguration_with_dkg;
+
+    struct RoundingResult has drop, store {
+        ideal_total_weight: u128,
+        weights: vector<u64>,
+        reconstruct_threshold_default_path: u128,
+        reconstruct_threshold_fast_path: Option<u128>,
+    }
+
+    struct CurEpochRounding has drop, key {
+        rounding: RoundingResult,
+    }
+
+    struct NextEpochRounding has drop, key {
+        rounding: RoundingResult,
+    }
+
+    /// When an async reconfig starts,
+    /// compute weights + threshold for the next validator set and store it on chain.
+    public(friend) fun on_reconfig_start(
+        framework: &signer,
+        stakes: vector<u64>,
+        secrecy_threshold_in_stake_ratio: FixedPoint64,
+        reconstruct_threshold_in_stake_ratio: FixedPoint64,
+        fast_secrecy_threshold_in_stake_ratio: Option<FixedPoint64>,
+    ) acquires NextEpochRounding {
+        let rounding = rounding(stakes, secrecy_threshold_in_stake_ratio, reconstruct_threshold_in_stake_ratio, fast_secrecy_threshold_in_stake_ratio);
+        if (exists<NextEpochRounding>(@aptos_framework)) {
+            move_from<NextEpochRounding>(@aptos_framework);
```

Comment:
I don't think it's worth passing the signer in just to initialize the resource. we should just do it in governance proposal and fallback to previous behavior if the resource doesn't exist?

Reply:
sure, that should work too